src/types.jl

```diff
@@ -183,11 +183,20 @@ function decompress_datastreambytes(compbytes, tkey)
             _decompress_lz4!(input_ptr, input_size, output_ptr, output_size)
         elseif cname == "ZL"
             # original: @view(uncomp_data[fufilled+1:fufilled+uncompbytes]) .= transcode(ZlibDecompressor, rawbytes)
-            input_ptr = pointer(rawbytes)
-            input_size = length(rawbytes)
-            output_ptr = pointer(uncomp_data) + fufilled
-            output_size = uncompbytes
-            _decompress_zlib!(input_ptr, input_size, output_ptr, output_size)
+
+            # If this is a one-shot decompression, use LibDeflate (faster)
+            if uncompbytes == tkey.fObjlen
+                # Need to ignore the 0x78 0xXX zlib header
+                decompress!(Decompressor(), uncomp_data, rawbytes[3:end], uncompbytes)
```

Comment:
That slicing makes a copy

Reply:
good catch

Reply:
Hmm thought I could do `@view(rawbytes[3:end])` but I get

```julia
ERROR: MethodError: no method matching decompress!(::LibDeflate.Decompressor, ::Vector{UInt8}, ::SubArray{UInt8, 1, Vector{UInt8}, Tuple{UnitRange{Int64}}, true}, ::Int64)
Closest candidates are:
  decompress!(::LibDeflate.Decompressor, ::Vector{UInt8}, ::Vector{UInt8}, ::Integer) at /Users/namin/.julia/packages/LibDeflate/JYL8P/src/LibDeflate.jl:153
  decompress!(::LibDeflate.Decompressor, ::Vector{UInt8}, ::Vector{UInt8}) at /Users/namin/.julia/packages/LibDeflate/JYL8P/src/LibDeflate.jl:163
```

consequence of Vector{Uint8} instead of AbstractArray?

Reply:
pop! or deleteat! Maybe

Reply:
`deleteat!(rawbytes, 1:2)` is perfect.

before and after:
```julia
julia> @btime sum(tf.nMuon)
  749.370 ms (12782 allocations: 581.54 MiB)
0x0000000008e67ad8

julia> @btime sum(tf.nMuon)
  707.929 ms (12162 allocations: 544.50 MiB)
0x0000000008e67ad8
```